feat(iam): add OIDC federations resources (#389)
selectel/terraform-provider-selectel: website/docs/r/iam_oidc_federation_v1.html.markdown

---
layout: "selectel"
page_title: "Selectel: selectel_iam_oidc_federation_v1"
sidebar_current: "docs-selectel-resource-iam-oidc-federation-v1"
description: |-
  Creates and manages OIDC Federation for Selectel products using public API v1.
---

# selectel\_iam\_oidc\_federation\_v1

Manages OIDC Federation for Selectel products using public API v1.
Selectel products support Identity and Access Management (IAM).

> For more information about federations, see the [official Selectel documentation](https://docs.selectel.ru/en/control-panel-actions/users-and-roles/federations/).

Reviewer: ```suggestion
For more information about federations, see the [official Selectel documentation](https://docs.selectel.ru/en/access-control/federations/).
```

Author: Fixed!

## Example Usage

```hcl
resource "selectel_iam_oidc_federation_v1" "federation_1" {

>   name                  = "federation name"

Reviewer: ```suggestion
  name                  = "Federation name"
```

Author: Fixed!

  alias                 = "federation-alias"

>   description           = "simple description"

Reviewer: ```suggestion
  description           = "Federation description"
```

Author: Fixed!

  issuer                = "https://idp.example.com/realms/master"
  client_id             = "my-client-id"
  client_secret         = "my-client-secret"
  auth_url              = "https://idp.example.com/realms/master/protocol/openid-connect/auth"
  token_url             = "https://idp.example.com/realms/master/protocol/openid-connect/token"
  jwks_url              = "https://idp.example.com/realms/master/protocol/openid-connect/certs"
  auto_users_creation   = true
  enable_group_mappings = true
  session_max_age_hours = 24
}
```

## Argument Reference

* `name` - (Required) Federation name.

* `alias` - (Optional) Federation alias.

* `description` - (Optional) Federation description.

> * `issuer` - (Required) Identifier of the credential provider.

Reviewer: ```suggestion
* `issuer` - (Required) Unique identifier of the credential provider.
```

Author: Fixed!

> * `client_id` - (Required) Client ID for OIDC authentication.

Reviewer: ```suggestion
* `client_id` - (Required) Unique identifier of the client for OIDC authentication.
```

Author: Fixed!

* `client_secret` - (Required, Sensitive) Client secret for OIDC authentication.

> * `auth_url` - (Required) Authorization endpoint URL.

Reviewer: ```suggestion
* `auth_url` - (Required) URL of the authorization endpoint used to authenticate users via the OIDC provider.
```

Author: Fixed!

> * `token_url` - (Required) Token endpoint URL.

Reviewer: ```suggestion
* `token_url` - (Required) URL of the token endpoint.
```

Author: Fixed!

> * `jwks_url` - (Required) JWKS endpoint URL.

Reviewer: ```suggestion
* `jwks_url` - (Required) URL of the JWKS endpoint with certificates used for token verification.
```

Author: Fixed!

> * `auto_users_creation` - (Optional) Enables automatic creation of users for this federation.

Reviewer: ```suggestion
* `auto_users_creation` - (Optional) Enables automatic user creation for this federation.
```

Author: Fixed!

> * `jwks_url` - (Required) URL of the JWKS endpoint with certificates used for token verification.

Reviewer: ```suggestion
* `jwks_url` - (Required) URL of the JSON Web Key Set (JWKS) endpoint with certificates used for token verification.
```

Author: Fixed!

* `enable_group_mappings` - (Optional) Enables group mappings for this federation.

> * `session_max_age_hours` - (Required) Session lifetime.

Reviewer: ```suggestion
* `session_max_age_hours` - (Required) Session lifetime in hours.
```

Author: Fixed!

## Attributes Reference

> * `account_id` - Selectel account ID. The account ID is in the top right corner of the [Control panel](https://my.selectel.ru/). Learn more about [Registration](https://docs.selectel.ru/en/control-panel-actions/account/registration/).

Reviewer: ```suggestion
* `account_id` - Selectel account ID. The account ID is in the top right corner of the [Control panel](https://my.selectel.ru/). Learn more about [Registration](https://docs.selectel.ru/en/account/registration/).
```

Author: Fixed!

> * `account_id` - Selectel account ID. The account ID is in the top right corner of the [Control panel](https://my.selectel.ru/). Learn more about [Registration](https://docs.selectel.ru/en/account/registration/).

Reviewer: ```suggestion
* `account_id` - Selectel account ID.
```

Author: Fixed!

## Import

You can import a federation:

```shell
export OS_DOMAIN_NAME=<account_id>
export OS_USERNAME=<username>
export OS_PASSWORD=<password>
terraform import selectel_iam_oidc_federation_v1.federation_1 <federation_id>
```

where:

> * `<account_id>` — Selectel account ID. The account ID is in the top right corner of the [Control panel](https://my.selectel.ru/). Learn more about [Registration](https://docs.selectel.ru/en/control-panel-actions/account/registration/).

Reviewer: ```suggestion
* `<account_id>` — Selectel account ID. The account ID is in the top right corner of the [Control panel](https://my.selectel.ru/). Learn more about [Registration](https://docs.selectel.ru/en/account/registration/).
```

Author: Fixed!

> * `<username>` — Name of the service user. To get the name, in the [Control panel](https://my.selectel.ru/iam/users_management/users?type=service), go to **Identity & Access Management** ⟶ **User management** ⟶ the **Service users** tab ⟶ copy the name of the required user. Learn more about [Service Users](https://docs.selectel.ru/en/control-panel-actions/users-and-roles/user-types-and-roles/).

Reviewer: ```suggestion
* `<username>` — Name of the service user. To get the name, in the [Control panel](https://my.selectel.ru/iam/service-users), go to **Account** ⟶ the **Service users** tab ⟶ copy the name of the required user. Learn more about [Service Users](https://docs.selectel.ru/en/access-control/user-types/).
```

Author: Fixed!

* `<password>` — Password of the service user.

> * `<federation_id>` — Unique identifier of the federation, for example, `abc1bb378ac84e1234b869b77aadd2ab`. To get the federation ID, use either [Control Panel](https://my.selectel.ru/iam/federations) or [Federations API](https://docs.selectel.ru/en/api/federations/).

Reviewer: ```suggestion
* `<federation_id>` — Unique identifier of the federation, for example, `abc1bb378ac84e1234b869b77aadd2ab`. Unique identifier of the federation, for example, `abc1bb378ac84e1234b869b77aadd2ab`. To get the federation ID, in the [Control Panel](https://my.selectel.ru/iam/federations), go to **Account** → **Federations** → copy the ID under the federation name.
```

Author: Fixed!

> * `<federation_id>` — Unique identifier of the federation, for example, `abc1bb378ac84e1234b869b77aadd2ab`. To get the federation ID, use either [Control Panel](https://my.selectel.ru/iam/federations) or [IAM API](https://developers.selectel.ru/docs/control-panel/iam/).

Reviewer: ```suggestion
* `<federation_id>` — Unique identifier of the federation, for example, `abc1bb378ac84e1234b869b77aadd2ab`. To get the federation ID, use either [Control Panel](https://my.selectel.ru/iam/federations) or [Federations API](https://docs.selectel.ru/en/api/federations/).
```

Author: Fixed!
